common/firmware/gpio.c

```diff
@@ -43,3 +43,10 @@ __attribute__((section( ".ccmram" )))
 bool gpio_is_set(void* port, uint16_t pin, uint8_t active_setting) {
     return HAL_GPIO_ReadPin(port, pin) == active_setting;
 }
+
+#ifdef ENABLE_CCMRAM
+__attribute__((section( ".ccmram" )))
+#endif
+uint8_t gpio_read(void* port, uint16_t pin) {
```

Comment:
we should prefer using `gpio_is_set`. the benefit of using `gpio_is_set` is that we can configure whether the pin is active high or active low, and then the core code doesn't have to care